node/commons/compatibility-layer.ts

```diff
@@ -340,14 +341,32 @@ const buildCommertialOffer = (
 }
 
 const getMarketingData = (segment?: SegmentData) => {
-  if (!segment || !segment.utm_campaign || !segment.utm_source) {
+  if(!segment?.utm_campaign && !segment?.utm_source && !segment?.utmi_campaign) {
     return
   }
 
-  return {
-    utmCampaign: segment.utm_campaign,
-    utmSource: segment.utm_source,
+  let marketingData = {}
+  if(segment?.utm_campaign) {
+    marketingData = {
+      utmCampaign: segment?.utm_campaign
+    }
+  }
+
+  if(segment?.utm_source) {
+    marketingData = {
+      ...marketingData,
+      utmSource: segment?.utm_source
+    }
```

Comment:
I think we always need `utm_source` when we use `utm_campaign`, so we need to ensure that we never send `utm_campaign` without the `utm_source`. 
Am I right, @hiagolcm? I remember you made an adjustment like this recently

Reply:
as far as I know, those values are separated, it's possible to have a campaign with one of them or all of them